### PR TITLE
SWQL-34 | Added eslint to project and fixed linting errors

### PR DESCRIPTION
**Pull request by laueian**
- Added eslint file an eslint ignore for sample files
- For many of eslint error I just added a disable comment because the recommended way didn't seem to make much sense
- At least now we have something checking our code before it is checked in

### CODE REVIEW

File: src/server/DBMetadata/classes/processedField.js [hunk 1]
```diff
@@ -51,7 +51,7 @@ class ProcessedField {
 		const { foreign_table_name, foreign_column_name } = tblCol;
 
 		// Active relationship assignment (primary key table defined before foreign key table)
-		if (lookup.hasOwnProperty(foreign_table_name)) {
+		if (Object.prototype.hasOwnProperty.call(foreign_table_name)) {
```
Comment (rob2244) [inline src/server/DBMetadata/classes/processedField.js]: This is unnecessarily verbose. Technically since all objects created inherit from Objects prototype, you can just use lookup.hasOwnProperty(foreign_table_name). There are cases where objects can be created that don't inherit from Objects prototype, but you have to actively try to do it.

File: src/server/DBMetadata/classes/processedField.js [hunk 2]
```diff
@@ -79,19 +78,15 @@ class ProcessedField {
 				[refIndex]: ref,
 			};
 
-			if (!toRef.hasOwnProperty(tblCol.foreign_table_name)) {
+			if (!Object.prototype.hasOwnProperty.call(tblCol.foreign_table_name)) {
```
Comment (rob2244) [inline src/server/DBMetadata/classes/processedField.js]: See my other comment

File: src/server/DBMetadata/classes/processedField.js [hunk 3]
```diff
@@ -79,19 +78,15 @@ class ProcessedField {
...
 
 				return;
 			}
 
-			if (
-				!toRef[tblCol.foreign_table_name].hasOwnProperty(
-					tblCol.foreign_column_name
-				)
-			) {
+			if (!Object[tblCol.foreign_table_name].prototype.hasOwnProperty.call(tblCol.foreign_column_name)) {
```
Comment (rob2244) [inline src/server/DBMetadata/classes/processedField.js]: See my other comment

File: src/server/DBMetadata/metadataProcessor.js [hunk 4]
```diff
@@ -66,10 +66,7 @@ function processMetadata (translateColumnType) {
 }
 
 function tableInToRef (toRef, tblCol) {
-	return (
-		toRef.hasOwnProperty(tblCol.table_name) &&
-    toRef[tblCol.table_name].hasOwnProperty(tblCol.column_name)
-	);
+	return Object.prototype.hasOwnProperty.call(tblCol.table_name) && Object[tblCol.table_name].prototype.hasOwnProperty.call(tblCol.column_name);
```
Comment (rob2244) [inline src/server/DBMetadata/metadataProcessor.js]: See my other comments on verbosity of Object.prototype.hasOwnProperty.call. Also I have no ides what the heck this is supposed to be doing: Object[tblCol.table_name].prototype.hasOwnProperty.call(tblCol.column_name) is that even valid javascript?

File: src/server/DBMetadata/mssql/msMetadataRetriever.js [hunk 5]
```diff
@@ -63,8 +63,7 @@ async function getDbPool (connString) {
 
 	if (poolCache[digest]) return poolCache[digest];
 
-	const pool = await new sql.ConnectionPool(connString).connect();
-	poolCache[digest] = pool;
+	const pool = await new sql.ConnectionPool(connString).connect().then(poolCache[digest] = pool);
```
Comment (rob2244) [inline src/server/DBMetadata/mssql/msMetadataRetriever.js]: This is incorrect, there's no reason to use .then, it's an async function that uses await. The original version is correct.

File: src/server/server.js [hunk 6]
```diff
@@ -1,14 +1,11 @@
...
 const { ipcMain } = electron;
-const dbController = require("./DBMetadata/pgsql/pgMetadataRetriever");
-const processMetaData = require("./DBMetadata/metadataProcessor");
 const generateGraphQL = require("./Generators/graphQLGenerator");
 const dbFactory = require("./dbFactory");
 const fs = require("fs");
 const JSZip = require("jszip");
 const path = require("path");
 const events = require("./events");
-const PgSqlProvider = require("./Generators/classes/pgSqlProvider");
```
Comment (rob2244) [inline src/server/server.js:11]: Why were these removed, aren't they needed?

**Review by rob2244 — CHANGES_REQUESTED** [review]

**Review by zahaan-jasani — APPROVED** [review]
